### PR TITLE
Code Cleanup and UI Work

### PR DESCRIPTION
**Pull request by Basileus1990**
.

### CODE REVIEW

File: src/main/java/net/darktree/MainWindow.java [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        dialog.setVisible(true);
+
+        String path = dialog.getDirectory() + dialog.getFile();
+        dialog.dispose();
+
+        if ("nullnull".equals(path)) {
+            return Optional.empty();
+        }
+
+        return Optional.of(path.substring(System.getProperty("user.dir").length() + 1));
```
Comment (magistermaks) [inline src/main/java/net/darktree/MainWindow.java:189]: What is this? :skull:
Do it like this:
```java
String filename = dialog.getFile();
String directory = dialog.getDirectory();

if (filename == null) {
    return Optional.empty();
}

// no idea what this is even supposed to do, why aren't you just using the path as-is?
return Optional.of(path.substring(System.getProperty("user.dir").length() + 1));
```
http://www.java2s.com/Code/JavaAPI/java.awt/FileDialoggetFile.htm

File: src/main/java/net/darktree/MainWindow.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+        JMenuItem hostMenuItem = new JMenuItem("Host");
+        MainWindow mainWindow = this;
+        hostMenuItem.addActionListener(e -> {
+            EventQueue.invokeLater(new Runnable() {
+                @Override
+                public void run() {
+                    new SharingPopup(SharingPopup.SharingType.HOST, mainWindow);
+                }
+            });
```
Comment (magistermaks) [inline src/main/java/net/darktree/MainWindow.java:217]: And here:
```java
EventQueue.invokeLater(() -> new SharingPopup(SharingPopup.SharingType.HOST, mainWindow));
```

**Comment by magistermaks** [comment]
I sense a general distaste for lambdas in your code :laughing: 
If you can fix those issues _(and the countless anonymous classes used instead of lambdas)_, if not i will merge and fix them myself later today

File: src/main/java/net/darktree/MainWindow.java (hunk 2 above)
Comment (Basileus1990) [reply src/main/java/net/darktree/MainWindow.java]: fixed

File: src/main/java/net/darktree/MainWindow.java [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+        // setting the save(ctrl+s) action
+        KeyStroke saveKeyStroke = KeyStroke.getKeyStroke(KeyEvent.VK_S, InputEvent.CTRL_DOWN_MASK);
+        getRootPane().getInputMap(JComponent.WHEN_IN_FOCUSED_WINDOW).put(saveKeyStroke, "saveCurrentFile");
+        getRootPane().getActionMap().put("saveCurrentFile", new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                saveCurrentFile();
+            }
+        });
```
[the remark this answers is not in the document]
Comment (Basileus1990) [reply src/main/java/net/darktree/MainWindow.java:82]: Lambdas cannot be used as Action

**Pull request merged**